velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -2747,6 +2747,61 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, arbitrateMemoryFromOtherOperator) {
   }
 }
 
+TEST_F(SharedArbitrationTest, joinBuildSpillError) {
+  const int kMemoryCapacity = 32 << 20;
+  // Set a small memory capacity to trigger hash join spill.
+  setupMemory(kMemoryCapacity, 0);
+
+  const int numVectors = 16;
+  std::vector<RowVectorPtr> vectors;
+  for (int i = 0; i < numVectors; ++i) {
+    vectors.push_back(newVector());
+  }
+
+  std::shared_ptr<core::QueryCtx> joinQueryCtx = newQueryCtx(kMemoryCapacity);
+
+  const int numDrivers = 4;
+  std::atomic<int> numAppends{0};
+  const std::string injectedErrorMsg("injected spillError");
+  SCOPED_TESTVALUE_SET(
+      "facebook::velox::exec::SpillState::appendToPartition",
+      std::function<void(exec::SpillState*)>([&](exec::SpillState* state) {
+        if (++numAppends != numDrivers) {
```

Comment:
Does this logic make first 3 operators to succeed in spilling, but 4-th (last) fail?

Reply:
I want to trigger the spilling error in the middle of the processing.

Reply:
Got it, but why not trigger it for all operators, i.e. unconditionally?

Reply:
Only the first exception is captured. I just want to throw in the middle

Reply:
I see... It is not guaranteed that appendToPartition is called only one per operator, is it? If that's not necessary the case then comparing numAppends with numDrivers is misleading. Would it be sufficient to just throw on second invocation?

Reply:
One operator could call appendToPartition multiple times so there is not guarantee there. We could change to call a second time.